### ISSUE TITLE
Latest jupyterlite doesn't work under Safari in macOS

### ISSUE THREAD

**Issue by ivanistheone** [jupyterlite/jupyterlite#944]
## Description

When commands in https://jupyterlite.readthedocs.io/en/latest/_static/lab/index.html 
using Safari 16.1 (16614.2.9.1.13, 16614) on macOS 11.7.1 (20G918), I get the following error:

<img width="863" alt="Screen Shot 2023-01-09 at 10 43 40 AM" src="https://user-images.githubusercontent.com/163966/211350055-412ea7e8-996e-4645-92a4-4b9e399004c2.png">

Note this was reported here: https://github.com/sympy/live/issues/16 when testing the sympy live shell

I just tested the pyodide console https://pyodide.org/en/stable/console.html and it seems to work fine in Safari, hence I believe it is a jupyterlite issue and opening here.

## Reproduce

0. Using Safari on a Mac
1. Go to https://jupyterlite.readthedocs.io/en/latest/_static/lab/index.html 
2. Open a new pyodide notebook
3. Try any command

## Expected behavior

Run the command

## Context

- JupyterLite version: `jupyterlite==0.1.0b17`

**Comment by bollwyvl** [comment]
Yep, might need a more robust polyfill for `setImmediate`, i suppose.

**Comment by henryiii** [comment]
Is there a known working version than can be pinned while this is being worked on? Ran into this with ~50 students in a workshop today.

**Comment by kgutwin** [comment]
I'm now having success with JupyterLite on Safari 17.0 (19616.1.27.211.1) under macOS Sonoma 14.0. 

Unfortunately, Safari 16.3 (17614.4.6.11.4) under macOS Monterey 12.6.3 still shows the "ReferenceError: Can't find variable: setImmediate" behavior.

**Comment by jtpio** [comment]
Thanks all for reporting.

I opened https://github.com/jupyterlite/jupyterlite/pull/1190 to investigate this. It seems like the `setImmediate` issue is indeed reproducible: https://github.com/jupyterlite/jupyterlite/pull/1190#issuecomment-1746730879

**Comment by jtpio** [comment]
Are folks still experiencing issues with the latest JupyterLite in Safari?

Trying the following two deployments with Safari 18.4:

- https://jupyterlite.readthedocs.io/en/latest/_static/lab/index.html (0.6.0 pre-releases)
- https://jupyterlite.github.io/demo/lab/index.html

Appears to be working fine:

![Image](https://github.com/user-attachments/assets/b778fe3c-0cd1-4e94-8cb7-3141735cb5a4)

**Comment by agriyakhetarpal** [comment]
I think there was a recent problem with iOS, but that was specific to Pyodide v0.27.1 and was resolved with Pyodide v0.27.3 – it didn't have anything to do with macOS or JupyterLite in general.

Both links are working for me on Safari version 18.4 (20621.1.15.11.10) with macOS Sequoia 15.4.1 (24E263) 👍🏻🚀

**Comment by jtpio** [comment]
Nice, thanks @agriyakhetarpal for checking!

So since the issue is more than 2 years old, let's assume this is now fixed. If some folks still have issues running JupyterLite with Safari, please feel free to comment here or open a new issue, thanks!
